pkg/firewall/firewallconfiguration_controller.go

```diff
@@ -0,0 +1,122 @@
+//
+//nolint:revive // We usually the name of the reconciled resource in the controller name.
+type FirewallConfigurationReconciler struct {
+	NftConnection *nftables.Conn
+	client.Client
+	Scheme         *runtime.Scheme
+	EventsRecorder record.EventRecorder
+}
+
+// NewFirewallConfigurationReconciler returns a new FirewallConfigurationReconciler.
+func NewFirewallConfigurationReconciler(cl client.Client, s *runtime.Scheme, er record.EventRecorder) (*FirewallConfigurationReconciler, error) {
+	nftConnection, err := nftables.New()
+	if err != nil {
+		return nil, fmt.Errorf("unable to create nftables connection: %w", err)
+	}
+	return &FirewallConfigurationReconciler{
+		NftConnection:  nftConnection,
+		Client:         cl,
+		Scheme:         s,
+		EventsRecorder: er,
+	}, nil
+}
+
+// cluster-role
+// +kubebuilder:rbac:groups=networking.liqo.io,resources=firewallconfigrations,verbs=get;list;watch;update;patch
+// +kubebuilder:rbac:groups=networking.liqo.io,resources=firewallconfigurations/status,verbs=get;list;watch;update;patch
+// +kubebuilder:rbac:groups=networking.liqo.io,resources=firewallconfigurations/finalizers,verbs=update
+
+// Reconcile manage FirewallConfigurations, applying nftables configuration.
+func (r *FirewallConfigurationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	fwcfg := &networkingv1alpha1.FirewallConfiguration{}
+	if err := r.Get(ctx, req.NamespacedName, fwcfg); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Infof("There is no firewallconfiguration %s", req.String())
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("unable to get the firewallconfiguration %q: %w", req.NamespacedName, err)
+	}
+
+	klog.Infof("Reconciling firewallconfiguration %s", req.String())
+
+	// Manage Finalizers and Table deletion.
+	if fwcfg.DeletionTimestamp.IsZero() {
+		if !ctrlutil.ContainsFinalizer(fwcfg, firewallConfigurationsControllerFinalizer) {
+			if err := r.ensureFirewallConfigurationFinalizerPresence(ctx, fwcfg); err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{}, nil
+		}
+	} else {
+		if ctrlutil.ContainsFinalizer(fwcfg, firewallConfigurationsControllerFinalizer) {
+			delTable(r.NftConnection, &fwcfg.Spec.Table)
+			if err := r.NftConnection.Flush(); err != nil {
+				return ctrl.Result{}, err
+			}
+			if err := r.ensureFirewallConfigurationFinalizerAbsence(ctx, fwcfg); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// Manage table creation and update.
+	addTable(r.NftConnection, &fwcfg.Spec.Table)
+
+	return ctrl.Result{}, r.NftConnection.Flush()
+}
+
+// SetupWithManager register the FirewallConfigurationReconciler to the manager.
+func (r *FirewallConfigurationReconciler) SetupWithManager(mgr ctrl.Manager, labels map[string]string) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&networkingv1alpha1.FirewallConfiguration{}, predicates(labels)).
+		Complete(r)
+}
```

Comment:
I think in this case you can use the LabelSelectorPredicate utility function. Example:
```suggestion
	filterByLabelsPredicate, err := predicate.LabelSelectorPredicate(metav1.LabelSelector{MatchLabels: labels})
	if err != nil {
		return err
	}
	return ctrl.NewControllerManagedBy(mgr).
		For(&networkingv1alpha1.FirewallConfiguration{}, builder.WithPredicates(filterBylabelsPredicate)).
		Complete(r)
}
```